Polish snap_build.py

bmw opened the pull request:
We hit this for the first time in a while recently and I wanted to try to make our tooling's messaging about it a little clearer.

EDIT: Also see https://github.com/certbot/certbot/pull/9584#issuecomment-1433532878.

alexzorin commented on tools/snap/build_remote.py:
```diff
@@ -78,7 +78,12 @@ def _execute_build(
+                print('Chroot problem encountered for build '
+                      f'{target} for {",".join(archs)}.\n'
+                      'Launchpad seems to be unable to recover from this '
+                      'state so were terminating the build.')
```
```suggestion
                      'state so we're terminating the build.')
```

bmw replied:
Nice catch!

bmw replied:
Nice catch!

bmw commented:
While fixing my typo/bad English, we happened to hit a "Chroot problem" failure! See the logs for the CI first attempt at https://dev.azure.com/certbot/certbot/_build/results?buildId=6416&view=results.

Looking at these logs, I noticed three things:

1. This message I added is sometimes printed many times because we're still processing output from `snapcraft`. See https://dev.azure.com/certbot/certbot/_build/results?buildId=6416&view=logs&j=f44d40a4-7318-5ffe-762c-ae4557889284&s=1dfbc15b-7d0f-52a9-b1da-b17592bf94f8&t=07786725-57f8-5198-4d13-ea77f640bd5c&l=565.
2. `snapcraft` is complaining that we should be using `--build-for` now instead of `--build-on`. See https://dev.azure.com/certbot/certbot/_build/results?buildId=6416&view=logs&j=f44d40a4-7318-5ffe-762c-ae4557889284&s=1dfbc15b-7d0f-52a9-b1da-b17592bf94f8&t=07786725-57f8-5198-4d13-ea77f640bd5c&l=472.
3. Us canceling the Certbot build due to a "Chroot problem" happened 3 times in 3 seconds which seems very unlikely. See https://dev.azure.com/certbot/certbot/_build/results?buildId=6416&view=logs&j=f44d40a4-7318-5ffe-762c-ae4557889284&s=1dfbc15b-7d0f-52a9-b1da-b17592bf94f8&t=07786725-57f8-5198-4d13-ea77f640bd5c&l=587. I looked at the builds on launchpad and I only saw one Certbot build. I think what's happening is [this code](https://github.com/certbot/certbot/blob/a3c9371dc5d3f51b1382da84acc3bb42bea8426b/tools/snap/build_remote.py#L137-L153) is causing the old build state to be reported so we error immediately.

I fixed all of these things in my follow up commits.

I hope you don't mind me pushing them to this PR too much. I initially started to fix them in a separate PR, but since (1) is made worse by the code I added here and fixing (3) separately causes merge conflicts, I decided to just do it all here.

alexzorin reviewed (approved):
Cool, these fixes make sense to me.
